tornado/test/web_test.py

```diff
@@ -700,11 +700,12 @@ def get(self):
             self.set_header("X-Foo", "foo\r\nX-Bar: baz")
             raise Exception("Didn't get expected exception")
         except ValueError as e:
-            if "Unsafe header value" in str(e):
-                self.finish(b"ok")
-            else:
+            if "Unsafe header value" not in str(e):
                 raise
 
+        self.set_header("X-Foo", "foo\tX-Bar: baz")
```

Comment:
This is nitpicky, but my first thought on seeing this test case is "why does it look like another header value after the tab? Is that some sort of injection attack I haven't seen?". Header injection is the reason `\r\n` are banned, but if we're expanding this handler to be the general test of `set_header`'s enforcement of valid header characters we should probably rename it and use examples that look less like injection attacks. 

Also, there should probably be a check in `test_header_injection` to assert that the valid header is readable from the client side.